# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/security/test/csp/test_report_uri_missing_in_report_only_header.html
# Commit: 8705284b50d4
# Full Hash: 8705284b50d4905110df9b1fab7f00d9d77d06e6
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-10-22 15:53:00
# Regressor Bug: 1584993
# File Overlap Count: 1
# Description:
#   Bug 1584993: Make CSP frame-ancestors work with fission enabled. r=jkt,farre,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49147
# ==============================================================================

diff -r 141755ff04e1 -r 8705284b50d4 dom/security/test/csp/test_report_uri_missing_in_report_only_header.html
--- a/dom/security/test/csp/test_report_uri_missing_in_report_only_header.html	Tue Oct 22 08:56:13 2019 +0000
+++ b/dom/security/test/csp/test_report_uri_missing_in_report_only_header.html	Tue Oct 22 08:53:47 2019 +0000
@@ -21,7 +21,16 @@
                           .getService(SpecialPowers.Ci.nsIStringBundleService);
 var localizer = stringBundleService.createBundle("chrome://global/locale/security/csp.properties");
 var warningMsg = localizer.formatStringFromName("reportURInotInReportOnlyHeader", [window.location.origin]);
+
+// Since Bug 1584993 we parse the CSP in the parent too, hence the
+// same error message appears twice in the console. 
+var callCleanUpOnce = false;
+
 function cleanup() {
+  if (callCleanUpOnce) {
+    return;
+  }
+  callCleanUpOnce = true;
   SpecialPowers.postConsoleSentinel();
   SimpleTest.finish();
 }